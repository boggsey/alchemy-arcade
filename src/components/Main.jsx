import React from 'react';
import { Switch, Route } from 'react-router-dom';

import HomePage from '../pages/HomePage';
import LoginPage from '../pages/LoginPage';
import PlayerPage from '../pages/PlayerPage';
import RegisterPage from '../pages/RegisterPage';
import RosterPage from '../pages/RosterPage';

const Main = () => (
  <main>
    <Switch>
      <Route exact path="/" component={HomePage} />
      <Route path="/login" component={LoginPage} />
      <Route path="/register" component={RegisterPage} />
      <Route path="/roster" component={RosterPage} />
      <Route path="/player/new" component={PlayerPage} />
      {/* <Route component={NoMatch} /> */}
    </Switch>
  </main>
);

export default Main;

