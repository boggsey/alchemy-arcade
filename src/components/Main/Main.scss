main {
  height: 100%;
  position: relative;
  display: flex;
  justify-content: center;
  align-items: center;
  overflow-x: scroll; 
}
