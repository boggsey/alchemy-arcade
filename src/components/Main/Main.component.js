import React from 'react';
import { Switch, Route } from 'react-router-dom';
import PropTypes from 'prop-types';
import PrivateRoute from '../Utils/PrivateRoute';
import HomePage from '../../pages/HomePage';
import LoginPage from '../../pages/LoginPage';
import PlayerPage from '../../pages/PlayerPage';
import RegisterPage from '../../pages/RegisterPage';
import RosterPage from '../../pages/RosterPage';
import ErrorPage from '../../pages/ErrorPage';
import './Main.scss';

const Main = (props) => {
  const { isAuthenticated } = props;
  return (
    <main>
      <Switch>
        <Route
          exact
          path="/"
          render={() => (
            <HomePage
              {...props}
              isAuthenticated={isAuthenticated}
            />
          )}
        />
        <Route path="/login" component={LoginPage} />
        <Route path="/register" component={RegisterPage} />
        <PrivateRoute
          path="/roster"
          component={RosterPage}
          isAuthenticated={isAuthenticated}
        />
        <PrivateRoute
          path="/player/new"
          component={PlayerPage}
          isAuthenticated={isAuthenticated}
        />
        <Route component={ErrorPage} />
      </Switch>
    </main>
  );
};

Main.propTypes = {
  isAuthenticated: PropTypes.bool.isRequired,
};

export default Main;
