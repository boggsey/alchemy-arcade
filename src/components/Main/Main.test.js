import React from 'react';
import { shallow } from 'enzyme';
import Main from './Main.component';

describe('Main', () => {
  it('Hasn\'t changed since the last snapshot', () => {
    const wrapper = shallow(<Main isAuthenticated />);

    expect(wrapper).toMatchSnapshot();
  });
});
