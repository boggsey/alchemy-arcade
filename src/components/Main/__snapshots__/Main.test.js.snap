// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Main Hasn't changed since the last snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Main
    isAuthenticated={true}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <Switch>
        <Route
          exact={true}
          path="/"
          render={[Function]}
        />
        <Route
          component={[Function]}
          path="/login"
        />
        <Route
          component={[Function]}
          path="/register"
        />
        <PrivateRoute
          component={[Function]}
          isAuthenticated={true}
          path="/roster"
        />
        <PrivateRoute
          component={[Function]}
          isAuthenticated={true}
          path="/player/new"
        />
        <Route
          component={[Function]}
        />
      </Switch>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <Route
            exact={true}
            path="/"
            render={[Function]}
          />,
          <Route
            component={[Function]}
            path="/login"
          />,
          <Route
            component={[Function]}
            path="/register"
          />,
          <PrivateRoute
            component={[Function]}
            isAuthenticated={true}
            path="/roster"
          />,
          <PrivateRoute
            component={[Function]}
            isAuthenticated={true}
            path="/player/new"
          />,
          <Route
            component={[Function]}
          />,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "exact": true,
            "path": "/",
            "render": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "component": [Function],
            "path": "/login",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "component": [Function],
            "path": "/register",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "component": [Function],
            "isAuthenticated": true,
            "path": "/roster",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "component": [Function],
            "isAuthenticated": true,
            "path": "/player/new",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "component": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": [Function],
    },
    "type": "main",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <Switch>
          <Route
            exact={true}
            path="/"
            render={[Function]}
          />
          <Route
            component={[Function]}
            path="/login"
          />
          <Route
            component={[Function]}
            path="/register"
          />
          <PrivateRoute
            component={[Function]}
            isAuthenticated={true}
            path="/roster"
          />
          <PrivateRoute
            component={[Function]}
            isAuthenticated={true}
            path="/player/new"
          />
          <Route
            component={[Function]}
          />
        </Switch>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <Route
              exact={true}
              path="/"
              render={[Function]}
            />,
            <Route
              component={[Function]}
              path="/login"
            />,
            <Route
              component={[Function]}
              path="/register"
            />,
            <PrivateRoute
              component={[Function]}
              isAuthenticated={true}
              path="/roster"
            />,
            <PrivateRoute
              component={[Function]}
              isAuthenticated={true}
              path="/player/new"
            />,
            <Route
              component={[Function]}
            />,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "exact": true,
              "path": "/",
              "render": [Function],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "component": [Function],
              "path": "/login",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "component": [Function],
              "path": "/register",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "component": [Function],
              "isAuthenticated": true,
              "path": "/roster",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "component": [Function],
              "isAuthenticated": true,
              "path": "/player/new",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "component": [Function],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": [Function],
      },
      "type": "main",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
